internal/compiler/typeloader.rs

```diff
@@ -590,6 +596,45 @@ impl TypeLoader {
             })
     }
 
+    /// Resolves include paths for the given .slint file by looking up the nearest `Cargo.toml`,
+    /// fetching its metadata, and finding any dependencies that depend on slint.
+    ///
+    /// This may get called multiple times for the same .slint file while looking up image
+    /// resources etc. referenced from the .slint file. The results are cached given that
+    /// fetching cargo metadata may be somewhat expensive for larger workspaces.
+    fn resolve_include_paths_for_file(&self, referencing_file: &Path) -> Option<Vec<PathBuf>> {
+        let manifest_file = manifest_directory(referencing_file)?.join("Cargo.toml");
+        if !manifest_file.exists() {
+            return None;
+        }
+
+        self.include_paths
+            .borrow_mut()
+            .entry(manifest_file.clone())
+            .or_insert_with(|| {
+                MetadataCommand::new().manifest_path(&manifest_file).exec().ok().map(|metadata| {
+                    let dependencies = metadata
+                        .packages
+                        .iter()
+                        .find(|p| p.manifest_path == manifest_file)
+                        .map(|p| p.dependencies.iter().map(|d| &d.name).collect::<HashSet<_>>())
+                        .unwrap_or_default();
+
+                    metadata
+                        .packages
+                        .iter()
+                        .filter(|p| {
+                            dependencies.contains(&p.name)
```

Comment:
This was a poor attempt to leave out irrelevant entries because the package list in cargo metadata includes the entire workspace. The problem might just go away with pre-computed include paths and explicit crate-prefixed imports, though.